e2e/e2e.sh

```diff
@@ -111,7 +111,7 @@ fi
 # here because then we can't easily get the SERVER_PID out of that subshell
 pushd "$BINARY_DIR"
 echo "Launching server from path $(pwd)"
-./ServerMain -i "$INDEX" -p 9099 -m 1GB -t --default-query-timeout 180s &> server_log.txt &
+./ServerMain -i "$INDEX" -p 9099 -m 1GB -t --default-query-timeout 500s &> server_log.txt &
```

Comment:
> Is this just a problem for debug builds, where the cancellation is rather slow?

Seems like it. Basically the only problem is the full index scan:

```
SELECT ?x ?y ?z WHERE {
  ?x ?y ?z .
  FILTER (?x = ?z)
}
```

where this seems to make a huge different in debug builds. Haven't actually compared it locally.

Regardless I don't think compiling without cancellation checks makes sense here because this would mean this particular thread sanitizer check would never check this particular code.

Reply:
Okay, then leave it for now. 
probably the solution is to have better implementation + cancellation for the full scans, but as they currently are really only feasible for small testing graphs, we can leave this (in fact this might be relevant for your next project:)